### PR TITLE
46170_UI_Design_using_Material_Design

### PR DESCRIPTION
**Pull request by shawon-sharma**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -23,4 +23,5 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile 'com.android.support:appcompat-v7:23.1.0'
     compile 'com.android.support:design:23.1.0'
+    compile 'com.android.support:recyclerview-v7:23.1.+'
```
Comment (shawon-sharma) [inline app/build.gradle]: sample comment

File: app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.ArrayList;
+import java.util.List;
+import  com.framgia.hrm.R;
+import  com.framgia.hrm.adapter.NavigationDrawerAdapter;
+import  com.framgia.hrm.model.NavDrawerItem;
+
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java]: please remove this blank line for beautiful code

File: app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    private RecyclerView recyclerView;
+    private ActionBarDrawerToggle mDrawerToggle;
+    private DrawerLayout mDrawerLayout;
+    private NavigationDrawerAdapter adapter;
+    private View containerView;
+    private static String[] titles = null;
+    private FragmentDrawerListener drawerListener;
+
+    public FragmentDrawer() {
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java]: blank line again

File: app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            navItem.setTitle(titles[i]);
+            data.add(navItem);
+        }
+        return data;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java]: blank line!

File: app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+        @Override
+        public void onTouchEvent(RecyclerView rv, MotionEvent e) {
+        }
+
+        @Override
+        public void onRequestDisallowInterceptTouchEvent(boolean disallowIntercept) {
+
+        }
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/FragmentDrawer.java]: blank line!

File: app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                              int dayOfMonth) {
+
+            editText.setText(String.valueOf(dayOfMonth) + "-" + String.valueOf(monthOfYear+1)
+                    + "-" + String.valueOf(year));
+        }
+    };
+
+    @Override
+    public void onAttach(Activity activity) {
+        super.onAttach(activity);
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java]: if not use this method, remove it please

File: app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    }
+
+    @Override
+    public void onDetach() {
+        super.onDetach();
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java]: same

**Comment by nvmanh** [comment]
I've checked this commit. Can you reformat code before pushing, i think reviewing code will be easier.
And try to remove un-user methods, params to update this commit

File: app/src/main/res/layout/fragment_registration.xml [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+            <Spinner
+                android:id="@+id/Spinner_dept"
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_alignParentRight="true"
+                android:layout_below="@id/Spinner"
+                android:layout_marginTop="20dp"
+                android:layout_toRightOf="@id/department"
+
```
Comment (nvmanh) [inline app/src/main/res/layout/fragment_registration.xml]: blank line!

File: app/src/main/java/com/framgia/hrm/activity/ViewFragment.java [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_view, container, false);
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/ViewFragment.java]: please remove this blank line

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+    @Override
+    public void onDrawerItemSelected(View view, int position) {
+        displayView(position);
+    }
+
+    private void displayView(int position) {
+        Fragment fragment = null;
+        String title = getString(R.string.app_name);
+        switch (position) {
+            case 0:
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: case 0, 1,2, 3 can you create constant value for these?

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        // Handle action bar item clicks here. The action bar will
+        // automatically handle clicks on the Home/Up button, so long
+        // as you specify a parent activity in AndroidManifest.xml.
+        int id = item.getItemId();
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: remove this blank line!

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+
+        drawerFragment = (FragmentDrawer)getSupportFragmentManager().findFragmentById(R.id.fragment_navigation_drawer);
+        drawerFragment.setUp(R.id.fragment_navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout), mToolbar);
+        drawerFragment.setDrawerListener(this);
+        displayView(0);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
+        fab.setOnClickListener(new View.OnClickListener()
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: can you re-format this, it's view not good!

File: app/src/main/res/layout/fragment_registration.xml (hunk 8 above)
Comment (phamxuanlu) [inline app/src/main/res/layout/fragment_registration.xml]: @shawon-sharma You should put all size as 20dp, 125dp... to values/dimens.xml instead using directly here.
Are you agree with me @nvmanh ? :D

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java (hunk 10 above)
Comment (shawon-sharma) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: what do you actually mean by creating constant value ?

File: app/src/main/java/com/framgia/hrm/activity/MainActivity.java [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+            default:
+                break;
+        }
+        if (fragment != null) {
+            FragmentManager fragmentManager = getSupportFragmentManager();
+            FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
+            fragmentTransaction.replace(R.id.container_body, fragment);
+            fragmentTransaction.commit();
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/MainActivity.java]: blank line!

File: app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(getContext(), R.array.activity_array, android.R.layout.simple_spinner_item);
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(adapter);
+
+        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+
+            @Override
+            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+                Toast.makeText(getContext(), position + "is selected", Toast.LENGTH_SHORT).show();
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java]: blank line!

File: app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+            public void onNothingSelected(AdapterView<?> parent) {}
+        });
+
+        //**************IMPLEMENTING SPINNER DROP DOWN FOR DEPARTMENT
+
+        Spinner spinner_dept = (Spinner) rootView.findViewById(R.id.Spinner_dept);
+        ArrayAdapter<CharSequence> adapter_dept = ArrayAdapter.createFromResource(getContext(), R.array.department_array, android.R.layout.simple_spinner_item);
+        adapter_dept.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner_dept.setAdapter(adapter_dept);
+
```
Comment (nvmanh) [inline app/src/main/java/com/framgia/hrm/activity/RegistrationFragment.java]: blank line!

File: app/src/main/res/layout/fragment_home.xml [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+    tools:context="com.framgia.hrm.activity.HomeFragment">
+    <TextView
+        android:id="@+id/label"
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_alignParentTop="true"
+        android:layout_marginTop="100dp"
+        android:gravity="center_horizontal"
+        android:text="HOME"
+        android:textSize="45dp"
```
Comment (nvmanh) [inline app/src/main/res/layout/fragment_home.xml]: please put it in dimension file, not hardcode

File: app/src/main/res/layout/fragment_home.xml [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+        android:textStyle="bold" />
+    <TextView
+        android:layout_width="fill_parent"
+        android:layout_height="wrap_content"
+        android:layout_below="@id/label"
+        android:layout_centerInParent="true"
+        android:layout_marginTop="10dp"
+        android:gravity="center_horizontal"
+        android:text="Edit fragment_home.xml to change the appearance"
+        android:textSize="12dp" />
```
Comment (nvmanh) [inline app/src/main/res/layout/fragment_home.xml]: dimension file please

**Comment by nvmanh** [comment]
- Please remove all blank lines
- Do not hardcode 12dp, 30sp.... in code, insert them to dimension file

**Comment by nvmanh** [comment]
OK!

**Pull request merged**